### PR TITLE
feat(integrations): settings list design touches

### PR DESCRIPTION
**Pull request by Chrissy**
This adds in a couple things from the original designs that I thought were good and got left out, notably:

* User tests showed that folks struggled to read the icons. I improved the "borderless" visual state for  buttons and used it to bump up the size.
* Added Clearer parent -> child relationship with smaller child avatar icons
* Added a separator between installation indicator and "learn more"
* Gave the "learn more" button a super low visual importance level

before:
image
after:
image

I also took this opportunity to tidy a few things up:

* moved animations into animations.jsx
* improved borderless styles in button to support text (all our current borderless buttons are icon-only)
* removed a ton of prop warnings from button (and generated x-million snapshot changes in the process)

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]
❤️ Just need to rebase and fix a bunch of stuff after Matte's refactor in https://github.com/getsentry/sentry/pull/9226

**Comment by Chrissy** [comment]
It isn't a _full_ deemphasis since I also made them larger, but I agree they do feel a tad less important. 

Ultimately it's a tricky balance, since we want to make the parent-child relationship clear but also keep the configuration button fairly prominent since a lot of good stuff is in there. I'll make them a touch darker and let's see how it feels in production.

**Comment by Chrissy** [comment]
@EvanPurkhiser heads up a gave a college try on swapping "learn more", but the circle looks strange right next to the bar. I also tried removing the bar, but then the circle just looks like a weird, accidentally large divider. Gonna leave as-is for now but we can of course revisit.

File: src/sentry/static/sentry/app/components/buttons/button.jsx [hunk 1]
```diff
@@ -234,18 +238,20 @@ const getLabelPadding = ({size, priority}) => {
...
     default:
-      return '12px 16px;';
+      return borderless ? '6px 10px' : '12px 16px';
   }
 };
 
-const ButtonLabel = styled(props => <Flex align="center" {...props} />)`
+const ButtonLabel = styled(({size, priority, borderless, ...props}) => (
+  <Flex align="center" {...props} />
+))`
```
[the remark this answers is not in the document]
Comment (Chrissy) [reply src/sentry/static/sentry/app/components/buttons/button.jsx]: wow that is an incredibly useful tip

**Comment by Chrissy** [comment]
@EvanPurkhiser minus the last comment you made yup it's ready! really struggling to keep the screenshots updated so gonna try to merge in sooner rather than later

**Comment by Chrissy** [comment]
but of course any additional feedback is always welcome!

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**